File: C++/Delete without head pointer.cpp
class Solution
{
    public:
    void deleteNode(Node *del_node)
    {
       Node *p=del_node;
       Node *q=p->next;
       while(q->next!=NULL){
           swap(p->data,q->data);
           p=q;
           q=q->next;
       }
       swap(p->data,q->data);
       p->next=NULL;
       delete(q);
    }

};

File: JAVA/Delete without head pointer.java
class Solution{
    void deleteNode(Node del_node){
        if(del_node.next==null)
            del_node=null;
        else{    
          del_node.data=del_node.next.data;
          del_node.next=del_node.next.next;
        }
    }
}
